CMakeLists.txt

```diff
@@ -559,6 +559,9 @@ if(SENTRY_BUILD_EXAMPLES)
 	add_executable(sentry_example examples/example.c)
 	target_link_libraries(sentry_example PRIVATE sentry)
 
+	target_compile_definitions(sentry PRIVATE SENTRY_PERFORMANCE_MONITORING)
```

Comment:
`SENTRY_BUILD_EXAMPLES` is I think the default if you directly compile the project (in contrast to compiling it as a sub-project), so IMO this goes a bit too far (enables it by default).
Better to make it a really opt-in flag, similar to the client-side stack trace feature.

Reply:
i don't think i have the knowledge to convert `SENTRY_PERFORMANCE_MONITORING` into a proper opt-in flag considering what i'm trying to prioritize right now:
- anybody building the example via `make build` should get perf monitoring out of the box, targeted towards other devs who may be contributing to perfmon 
- unit tests should have perf monitoring enabled by default

i could create a separate `make` command to build the project with the performance monitoring flag enabled. this just raises other build issues with the unit tests though, as they're missing symbols locked behind the feature flag. 

i'm open to any help to actually make this work, but right now automatically turning the flag on when `SENTRY_BUILD_EXAMPLES` is on appears to be my best bet to getting this working. the long term plan is to either remove the flag entirely or to fully commit to it anyways, so this should be a good enough temporary solution i think.

Reply:
i'll leave the PR open until the start of the work day tomorrow just to get some follow-up feedback on this and to see if there are any deal breakers in the current diff. otherwise, i'll be merging this in.